src/Interpreters/Streaming/CachedBlockMetrics.h

```diff
@@ -11,25 +13,30 @@ namespace Streaming
 {
 struct CachedBlockMetrics
 {
-    size_t current_total_blocks = 0;
-    size_t current_total_bytes = 0;
+    size_t total_rows = 0;
     size_t total_blocks = 0;
-    size_t total_bytes = 0;
+    size_t total_metadata_bytes = 0;
+    size_t total_data_bytes = 0;
     size_t gced_blocks = 0;
 
+    ALWAYS_INLINE size_t totalBytes() const { return total_metadata_bytes + total_data_bytes; }
+
     std::string string() const
     {
         return fmt::format(
-            "total_bytes={} total_blocks={} current_total_bytes={} current_total_blocks={} gced_blocks={}",
-            total_bytes,
+            "total_rows={} total_bytes={} total_blocks={} gced_blocks={} (total_metadata_bytes:{} total_data_bytes:{})",
+            total_rows,
+            totalBytes(),
             total_blocks,
-            current_total_bytes,
-            current_total_blocks,
-            gced_blocks);
+            gced_blocks,
+            total_metadata_bytes,
+            total_data_bytes);
     }
 
-    void serialize(WriteBuffer & wb) const;
-    void deserialize(ReadBuffer & rb);
+    /// [Legacy] We don't need to serialize this anymore on new impl, since the metrics is volated. will update it back during recover 
+    static constexpr VersionType HAS_STATE_MAX_VERSION = 1;
```

Comment:
What's `HAS_STATE_MAX_VERSION`, didn't get what it means ? It is for metrics ?

Reply:
It is used to indicate the max version that needs to continue to do state serialization/deserialization. use `NEED_SERDE_MAX_VERSION` ?